app/src/components/GameRoutes.tsx

```diff
@@ -15,14 +26,58 @@ function CurrentPlayerProvider(props: {
   joinCode: string
   children: React.ReactNode
 }) {
+  const currentAuth = React.useContext(CurrentAuthContext)
+  const [redirectRoute, setRedirectRoute] = React.useState("")
+  const [, updateState] = React.useState()
+  const forceUpdate = React.useCallback(() => updateState({}), [])
+
+  // https://github.com/apollographql/react-apollo/issues/3505#issuecomment-568112061
+  const [skipCallback, setSkipCallback] = React.useState(false)
+
+  const [joinGame] = useJoinGameMutation()
+  const [loadGame] = useLazyQuery(GameByJoinCodeDocument, {
+    variables: { joinCode: props.joinCode.toLocaleUpperCase() },
+    onCompleted: async data => {
+      if (skipCallback) {
+        return
+      }
+      if (data && data.games[0]) {
+        const registration = await joinGame({
+          variables: {
+            gameId: data.games[0].id,
+            clientUuid: clientUuid()
+          }
+        })
+        if (registration.data?.joinGame) {
+          await currentAuth.setJwtToken(registration.data.joinGame.jwt_token)
+        }
+        setSkipCallback(true)
+        forceUpdate()
```

Comment:
force an update, and reflow this component - which will then call `useCurrentPlayerQuery` again and we'll have a valid JWT this time (no errors)

avoids using another lazyQuery which loses some typing / generally harder to work with

Reply:
ah, nice, didn't know this was possible 👍